nano/node/node.cpp

```diff
@@ -139,13 +140,15 @@ nano::node::node (std::shared_ptr<boost::asio::io_context> io_ctx_a, uint16_t pe
 }
 
 nano::node::node (std::shared_ptr<boost::asio::io_context> io_ctx_a, std::filesystem::path const & application_path_a, nano::node_config const & config_a, nano::work_pool & work_a, nano::node_flags flags_a, unsigned seq) :
```

Comment:
A reference to io_ctx is still unnecessarily passed externally. Removing it now would add noise. Such cleanup can be easily done in a subsequent PR.